lib/pacemaker/pcmk_sched_colocation.c

```diff
@@ -1152,16 +1144,12 @@ pcmk__apply_coloc_to_scores(pe_resource_t *dependent,
                             const pe_resource_t *primary,
                             const pcmk__colocation_t *colocation)
 {
-    const char *attribute = CRM_ATTR_ID;
+    const char *attribute = colocation->node_attribute;
```

Comment:
It is surprising, but I suppose defaulting to node ID or name doesn't matter as long as there's a one-to-one mapping between node IDs and names. Definitely a good idea to make it consistent.